File: 20-valid-parentheses/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> sStack=new Stack<Character>();
        
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='('){
                sStack.push(')');
            }else if(s.charAt(i)=='{'){
                sStack.push('}');
            }else if(s.charAt(i)=='['){
                sStack.push(']');
            }else if(sStack.empty()==true){
                return false;
            }else if(s.charAt(i)!=sStack.peek()){
                return false;
            }else{
                sStack.pop();
            }
        }
        
        return sStack.empty();
    }
}

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        stack=[]

        for i in s:
            if i=='(':
                stack.append(')')
            elif i=='[':
                stack.append(']')
            elif i=='{':
                stack.append('}')
            elif not stack:
                return False
            elif i!=stack[-1]:
                return False
            else:
                stack.pop()
        
        return not stack